specs/experimental/op-contracts-manager.md

```diff
@@ -20,146 +20,271 @@ of governance approved [contract releases] can be found on the
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 **Table of Contents**
 
+- [Overview](#overview)
+- [Getter Methods](#getter-methods)
 - [Deployment](#deployment)
-- [Interface](#interface)
-  - [`Proxy.sol`](#proxysol)
-  - [`deploy`](#deploy)
-  - [Getter Methods](#getter-methods)
-- [Implementation](#implementation)
-  - [Batch Inbox Address](#batch-inbox-address)
-  - [Contract Deployments](#contract-deployments)
+  - [Interface](#interface)
+    - [`deploy`](#deploy)
+  - [Implementation](#implementation)
+    - [Batch Inbox Address](#batch-inbox-address)
+    - [Contract Deployments](#contract-deployments)
+- [Upgrading](#upgrading)
+  - [Interface](#interface-1)
+    - [`upgrade`](#upgrade)
+  - [Implementation](#implementation-1)
+    - [`NewChainConfig` struct](#newchainconfig-struct)
+    - [Requirements on the OP Chain contracts](#requirements-on-the-op-chain-contracts)
+- [Adding game types](#adding-game-types)
+  - [Interface](#interface-2)
+    - [`addGameType`](#addgametype)
+  - [Implementation](#implementation-2)
 - [Security Considerations](#security-considerations)
   - [Chain ID Source of Truth](#chain-id-source-of-truth)
   - [Chain ID Frontrunning](#chain-id-frontrunning)
   - [Chain ID Value](#chain-id-value)
   - [Proxy Admin Owner](#proxy-admin-owner)
-  - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+  - [Safely using `DELEGATECALL`](#safely-using-delegatecall)
+  - [Atomicity of upgrades](#atomicity-of-upgrades)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## Deployment
+## Overview
 
-The OP Contracts Manager is a proxied contract deployed at `0xTODO`. It can be deployed as follows:
+The OP Contracts Manager refers to a series of contracts, of which a new singleton is deployed
```

Comment:
It could be good to define upgrade controller early in the doc

Reply:
7404d958896c